samtranslator/model/events.py

```diff
@@ -16,3 +23,27 @@ class EventsRule(Resource):
     }
 
     runtime_attrs = {"rule_id": lambda self: ref(self.logical_id), "arn": lambda self: fnGetAtt(self.logical_id, "Arn")}
+
+    def __init__(
+        self,
+        logical_id: Optional[Any],
+        relative_id: Optional[str] = None,
+        depends_on: Optional[List[str]] = None,
+        attributes: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        super().__init__(logical_id, relative_id, depends_on, attributes)
+
+        if len(self.logical_id) > EVENT_RULE_ID_MAX_LENGTH:
+            # Truncate logical id to satisfy the EVENT_RULE_ID_MAX_LENGTH limit
+            self.logical_id = (
+                self.logical_id[: (EVENT_RULE_ID_MAX_LENGTH - len(EVENT_RULE_ID_EVENT_SUFFIX))]
+                + EVENT_RULE_ID_EVENT_SUFFIX
+            )
```

Comment:
Shouldn't this call the function underneath?

Reply:
No, because the original implementation uses `logical_id`, not `logical_id + "Events"`. If we use the fucntion below, it could be a breaking change.

Reply:
I would really enjoy a

```python
def _truncate_with_suffix(s: str, length: int, suffix: str):
    return s[: length - len(suffix)] + suffix
```

🥺

The logic is simple but it's deceivingly similar yet formatting is different it's hard to pattern match at a glance to figure out the difference.